File: Inventario/DataAccess/Models/Detail.cs
using System;

namespace Data
{
    public class Detail: Entity
    {
        public Guid SaleId { get; set; }
        public Sale Sale { get; set; }
        public Guid ProductId { get; set; }
        public Product Product { get; set; }
        public decimal UnitPrice { get; set; }
        public int Quantity { get; set; }
    }
}

File: inventario_frontend/src/models/detail.ts
import { Product } from "./product";

export class DetailForCreation {
  public productId: string | undefined = undefined;
  public quantity: number | undefined = undefined;
}

export class Detail extends DetailForCreation {
  public id!: string;
  public saleId!: string;
  public product: Product | undefined = undefined;
  public unitPrice!: number;
}
